src/mono/wasm/build/WasmApp.targets

```diff
@@ -135,6 +135,8 @@
 
     <!-- by default, package assemblies as webcil -->
     <WasmEnableWebcil Condition="'$(WasmEnableWebcil)' == ''">true</WasmEnableWebcil>
+
+    <WasmEnableThreads Condition="'$(WasmEnablePerfTracing)' == 'true' or '$(MonoWasmBuildVariant)' == 'multithread'">true</WasmEnableThreads>
```

Comment:
I dislike this a lot, it would be better if enabling perf tracing without enabling threads caused the build to fail with an `<Error>`. Otherwise you will get surprising differences in behavior (no jiterpreter, no snapshots) by turning on a feature that seems totally unrelated

Reply:
@lambdageek @lewing let's remove "perf tracing" flavor. I think we don't have bandwidth to support it.

Reply:
Could we make it automatically enabled whenever threads are on? is it expensive?

Reply:
simplified by removing perftracing flavor

Reply:
I think we can kill the `MonoWasmBuildVariant=perftrace` configuration (I don't htink it was seeing much use, and it's kind of strange).

I think we should keep the diagnostic server and event pipe support if someone is building the multi-threaded runtime.   (ie: if `MonoWasmBuildVariant=true` then we should enable `FeaturePerfTracing` when building the multi-threaded CoreLib).

`WasmEnablePerfTracing` is a user-visible toggle. I think the way things work today is that it is exclusive with `WasmEnableThreads` - a user project can set one or the other, but not both.  I think if a user project sets perftracing to true or it enables threads, the eventpipe and diagnostic server code is included in the app.  
I'm not sure if we should just ignore `WasmEnablePerfTracing` or just always error out if someone tries to set it to true and point them at `WasmEnableThreads` instead.

I think in a future .NET release it would be good to build a single-threaded version of eventpipe and the diagnostic server.